lte/protos/subscriberauth.proto

```diff
@@ -0,0 +1,98 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+//
+// This is a proxy for Diamater application S6a/S6d 3GPP 29.272
+// It is watered down for use with OpenAirInterface EPC with extensibility in mind
+//
+// For details about fields see:
+// http://www.etsi.org/deliver/etsi_ts/129200_129299/129272/10.09.00_60/ts_129272v100900p.pdf
+
+syntax = "proto3";
+
+package magma.lte;
+option go_package = "magma/lte/cloud/go/protos";
+
+// ErrorCode reflects Experimental-Result values which are 3GPP failures
+// to be processed by EPC. Diameter Base Protocol errors are reflected in gRPC status code
+enum M5GErrorCode {
+    UNDEFINED                = 0;
+    // Default success code
+    MULTI_ROUND_AUTH         = 1001;
+    SUCCESS                  = 2001;
+    LIMITED_SUCCESS          = 2002;
+    COMMAND_UNSUPORTED       = 3001;
+    UNABLE_TO_DELIVER        = 3002;
+    REALM_NOT_SERVED         = 3003;
+    TOO_BUSY                 = 3004;
+    LOOP_DETECTED            = 3005;
+    REDIRECT_INDICATION      = 3006;
+    APPLICATION_UNSUPPORTED  = 3007;
+    INVALIDH_DR_BITS         = 3008;
+    INVALID_AVP_BITS         = 3009;
+    UNKNOWN_PEER             = 3010;
+    AUTHENTICATION_REJECTED  = 4001;
+    OUT_OF_SPACE             = 4002;
+    ELECTION_LOST            = 4003;
+    AUTHORIZATION_REJECTED   = 5003;
+    // Permanent Failures 7.4.3
+    USER_UNKNOWN             = 5001;
+    UNKNOWN_SESSION_ID       = 5002;
+    UNKNOWN_EPS_SUBSCRIPTION = 5420;
+    RAT_NOT_ALLOWED          = 5421;
+    ROAMING_NOT_ALLOWED      = 5004;
+    EQUIPMENT_UNKNOWN        = 5422;
+    UNKOWN_SERVING_NODE      = 5423;
+    // Transient Failures 7.4.4
+    AUTHENTICATION_DATA_UNAVAILABLE = 4181;
+}
+
+// Authentication Information Request (Section 7.2.5)
+message M5GAuthenticationInformationRequest {
+    // Subscriber identifier
+    string user_name = 1;
+    // Visted site identifier
+    bytes visited_plmn = 2;
+    // Number of vectors to request in response
+    uint32 num_requested_eutran_vectors = 3;
+    // Indicates to the HSS the values are requested for immediate attach
+    bool immediate_response_preferred = 4;
+    // Concatenation of RAND and AUTS in the case of a resync attach case
+    bytes resync_info = 5;
+    // Number of UTRAN/GERAN vectors to request in response (optional)
+    uint32 num_requested_utran_geran_vectors = 6;
+    // UTRAN/GERAN Resync Info (optional)
+    bytes utran_geran_resync_info = 7;
+}
+
+// Authentication Information Answer (Section 7.2.6)
+message M5GAuthenticationInformationAnswer {
+    // 3GPP TS 29.272, 7.3.18 E-UTRAN-Vector
+    // For details about fields read 3GPP 33.401
+    message EUTRANVector {
+        bytes rand = 1;
+        bytes xres = 2;
+        bytes autn = 3;
+        bytes kasme = 4;
+        bytes ck = 5;
+        bytes ik = 6;
+    }
```

Comment:
This needs to be redefined as "5G Authentication Vector: authentication data consisting of RAND, AUTN, HXRES*, and KSEAF". You can define a message as FiveGAuthVector.

Note that as part of this change, you need to migrate the key generation up to SEAF from AMF to subscriberdb side.

Reply:
We will include the above in "PR1 for 5G Authentication"

Reply:
The comment is addressed and #8051 is integrated.